pkg/ui/workspaces/db-console/src/redux/alerts.ts

```diff
@@ -707,6 +695,105 @@ export const licenseTypeSelector = createSelector(
   data => licenseTypeNames.get(data.LicenseType) || "None",
 );
 
+export const licenseUpdateDismissedLocalSetting = new LocalSetting(
+  "license_update_dismissed",
+  localSettingsSelector,
+  moment(0),
+);
+
+const licenseUpdateDismissedPersistentLoadedSelector = createSelector(
+  (state: AdminUIState) => state.uiData,
+  uiData => uiData && has(uiData, LICENSE_UPDATE_DISMISSED_KEY),
+);
+
+const licenseUpdateDismissedPersistentSelector = createSelector(
+  (state: AdminUIState) => state.uiData,
+  uiData => {
+    return (
+      (uiData &&
+        uiData[LICENSE_UPDATE_DISMISSED_KEY] &&
+        uiData[LICENSE_UPDATE_DISMISSED_KEY].data &&
+        moment(uiData[LICENSE_UPDATE_DISMISSED_KEY].data)) ||
+      moment(0)
+    );
+  },
+);
+
+export const licenseUpdateNotificationSelector = createSelector(
+  licenseTypeSelector,
+  licenseUpdateDismissedLocalSetting.selector,
+  licenseUpdateDismissedPersistentSelector,
+  licenseUpdateDismissedPersistentLoadedSelector,
+  (
+    licenseType,
+    licenseUpdateDismissed,
+    licenseUpdateDismissedPersistent,
+    licenseUpdateDismissedPersistentLoaded,
+  ): Alert => {
+    // If customer has Enterprise license they don't need to worry about this.
+    if (licenseType === "Enterprise") {
+      return undefined;
+    }
+
+    // If the notification has been dismissed based on the session storage
+    // timestamp, don't show it.
+    if (licenseUpdateDismissed.isAfter(moment(0))) {
+      return undefined;
+    }
+
+    // If the notification has been dismissed based on the uiData
+    // storage in the cluster, don't show it. Note that this is
+    // different from how version upgrade notifications work, this one
+    // is dismissed forever and won't return even if you upgrade
+    // further or time passes.
+    if (
+      licenseUpdateDismissedPersistentLoaded &&
+      licenseUpdateDismissedPersistent ) {
```

Comment:
I'm a little confused by this case because based on the selector right above this one, doesn't `licenseUpdateDismissedPersistent` always return a moment as long as there is uiData loaded? Why don't we need to do the `isAfter(moment(0))` check here?

Reply:
you are correct. added.